### PR TITLE
Support log record context in JSON messages

### PR DESCRIPTION
**Pull request by aarb1**

### CODE REVIEW

**Comment by aarb1** [comment]
#build

**Comment by aarb1** [comment]
Merge conflicts removed, but manually untested as of right now.

**Comment by aarb1** [comment]
Another build on EBC2: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewDefinition&id=_NzLdELuvEeagX9uhHryupg

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java [hunk 1]
```diff
@@ -228,4 +236,90 @@ protected static String jsonifyTags(String[] tags) {
...
+            if (value.equals(TRUE_BOOL) || value.equals(FALSE_BOOL)) {
+                isValidExt = true;
+            }
+        } else {
+            isValidExt = true;
+        }
+        return isValidExt;
+    }
+
+    public static boolean verifyIntValue(String value) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java]: should probably be private static boolean

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java [hunk 2]
```diff
@@ -228,4 +236,90 @@ protected static String jsonifyTags(String[] tags) {
...
+                continue;
+            } else {
+                isValid = false;
+                break;
+            }
+        }
+        return isValid;
+    }
+
+    public static boolean verifyFloatValue(String s) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java]: should probably be private static boolean

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java [hunk 3]
```diff
@@ -228,4 +236,90 @@ protected static String jsonifyTags(String[] tags) {
...
+            } else {
+                return false;
+            }
+
+        }
+
+        return isValid;
+    }
+
+    protected static boolean checkIfExtNum(String extKey) {
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java]: should change this to checkIfExtRequiresQuotes and add BOOL_SUFFIX to the list below and return *false* (NOT *true*) for int/float/bool

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonUtils.java [hunk 4]
```diff
@@ -230,6 +233,22 @@ else if (key.equals(LogFieldConstants.MESSAGE)) {
...
+                kvpl = (KeyValuePairList) p;
+                extensions = kvpl.getKeyValuePairs();
+                boolean isExtNum = false;
+                for (KeyValuePair k : extensions) {
+                    String extKey = k.getKey();
+                    String extValue = k.getValue();
+                    // Checking the value of the extension KeyValuePair to make sure the suffix is correct
+                    boolean isValidExt = CollectorJsonHelpers.checkExtSuffixValidity(k);
+                    if (isValidExt) {
+                        isExtNum = CollectorJsonHelpers.checkIfExtNum(extKey);
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonUtils.java]: should change this to isExtRequiresQuotes

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonUtils1_1.java [hunk 5]
```diff
@@ -341,6 +344,22 @@ else if (key.equals(LogFieldConstants.MESSAGE)) {
...
+                kvpl = (KeyValuePairList) p;
+                extensions = kvpl.getKeyValuePairs();
+                boolean isExtNum = false;
+                for (KeyValuePair k : extensions) {
+                    String extKey = k.getKey();
+                    String extValue = k.getValue();
+                    // Checking the value of the extension KeyValuePair to make sure the suffix is correct
+                    boolean isValidExt = CollectorJsonHelpers.checkExtSuffixValidity(k);
+                    if (isValidExt) {
+                        isExtNum = CollectorJsonHelpers.checkIfExtNum(extKey);
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonUtils1_1.java]: same comments from CollectorJsonUtils apply here

**Comment by aarb1** [comment]
#build

**Pull request closed without merging**